homeassistant/components/efesto/climate.py

```diff
@@ -0,0 +1,218 @@
+"""Support for Efesto heating devices."""
+import logging
+import socket
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.climate import ClimateDevice, PLATFORM_SCHEMA
+from homeassistant.components.climate.const import (
+    SUPPORT_ON_OFF, ATTR_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_URL, CONF_PASSWORD, CONF_USERNAME, CONF_DEVICE,
+    PRECISION_WHOLE, STATE_OFF, STATE_ON, TEMP_CELSIUS
+)
+
+REQUIREMENTS = ['efestoclient==0.0.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_DEVICE): cv.string,
+})
+
+ATTR_DEVICE_STATUS = 'device_status'
+ATTR_SMOKE_TEMP = 'smoke_temperature'
+ATTR_REAL_POWER = 'real_power'
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Efesto device."""
+    url = config.get(CONF_URL)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    device = config.get(CONF_DEVICE)
+
+    return _setup_efesto(
+        url, username, password, device, config, add_entities
+    )
+
+
+def _setup_efesto(url, username, password, device, config, add_entities):
+    """Set up the rounding function."""
+    from efestoclient import EfestoClient
+
+    client = EfestoClient(url, username, password, device, False)
+
+    try:
+        add_entities(
+            [EfestoHeatingDevice(client)],
+            True
+        )
+    except socket.error:
```

Comment:
Constructor can't raise and you should not update exceptions leak out from update. So this try/except should be removed.

Reply:
Ok I'll remove it, thanks.